Assets/AltTester/Editor/AltTestRunner.cs

```diff
@@ -71,7 +71,7 @@ public static void RunTests(TestRunMode testMode)
             }
             if (AltTesterEditorWindow.EditorConfiguration.createXMLReport)
             {
-                if (AltTesterEditorWindow.EditorConfiguration.xMLFilePath.Equals(""))
+                if (AltTesterEditorWindow.EditorConfiguration.xMLFilePath.Equals("") || AltTesterEditorWindow.EditorConfiguration.xMLFilePath.Substring(AltTesterEditorWindow.EditorConfiguration.xMLFilePath.Length - 3, AltTesterEditorWindow.EditorConfiguration.xMLFilePath.Length).Equals(".xml"))
```

Comment:
* To get the extension of a file in C#, you can use the `GetExtension(String)` method. Check out the documentation for more information: https://learn.microsoft.com/en-us/dotnet/api/system.io.path.getextension?view=net-7.0#system-io-path-getextension(system-string)
* For this to work you want to check if the file doesn't have the `.xml` extension.
* However, I think it would be better to check if the path is a directory, and it it is a directory create a `.xml` file inside the directory.